st2client/st2client/client.py

```diff
@@ -129,115 +130,194 @@ def __init__(
 
         self.api_key = api_key
 
+        if basic_auth:
+            if len(basic_auth.split(":")) != 2:
```

Comment:
This prohibits passwords with a `:` in them. We should either document the password requirements somewhere or update the operator to >=.

Reply:
Good catch, will use left split with limit of 1 (this should work as long as the username doesn't contain a colon).

Reply:
e67d40e9d